File: Data Structures & Algorithms/number-of-one-bits/submission-4.java
class Solution {
    public int hammingWeight(int n) {
        int count = 0;
        for (int i = 0; i < 32; i++) {
            count += (n >> i) & 1;
        }
        return count;
    }
}

// class Solution:
//     def hammingWeight(self, n: int) -> int:
//         count = 0
//         for i in range(32):
//             count += (n >> i) & 1
//         return count
        

File: Data Structures & Algorithms/gas-station/submission-4.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        start = tank = total_gas = 0
        for i in range(len(gas)):
            net = gas[i] - cost[i]
            tank += net
            total_gas += net
            if tank < 0:
                start = i + 1
                tank = 0
        if total_gas < 0:
            return -1
        return start
